src/main/webapp/script.js

```diff
@@ -64,69 +64,78 @@ async function onSubmitUploadImageForm() {
-      }).catch((error) => {
-        console.warn(error);
-        return new Response(JSON.stringify({
-          code: error.response.status,
-          message: `Failed to fetch ${imageUploadUrl}`,
-        }));
-      });
+    method: 'POST',
+    body: formData,
+  }).catch((error) => {
+    console.warn(error);
+    return new Response(JSON.stringify({
+      code: error.response.status,
+      message: `Failed to fetch ${imageUploadUrl}`,
+    }));
+  });
 
   if (!response.ok) {
     return Promise.reject(response);
   }
 
-  // The request returns a JSON array with the shopping query used to search on Google Shopping and
-  // the data about each product from the Google Shopping results page.
-  const data = await response.json();
-  const shoppingQuery = data[0]; 
-  const products = data[1]; 
+  // The request returns a JSON object with the all the shopping queries used to search on Google Shopping and
+  // the data about each product for each query from the Google Shopping results page.
+  const shoppingResults = await response.json();
 
-  // Empty the prompt container and add the {@code products} content into the page.
+  // Empty the prompt container and add the query and its resultant content into the page.
   $('#search-loading-prompt').empty();
   $('#loading-gif-prompt').addClass('hidden');
 
-  // Show the user the shopping query built to search on Google Shopping.
   $('#shopping-query-display-container').removeClass('hidden');
-  $('#shopping-query').text(shoppingQuery);
+  // So that the queries list starts at a new line
+  $('#shopping-query').append('<br />');
 
-  appendShoppingResults(products);
+  appendShoppingResults(shoppingResults);
 }
 
 /**
- * Integrates product results from Google Shopping into the web page. 
+ * Integrates product results from Google Shopping and their respective queries into the web page.
  */
-async function appendShoppingResults(products) {
-  // Integrate the products into the web page.
-  products.forEach(product => {
-    // Create an HTML node for the item container.
-    let $productContainer = $('<div>', {class: 'col-md-4'});
-
-    // Get the HTML content for the container.
-    const productElementHTML = getProductElementHTML(product.title,
-                                                     product.imageLink,
-                                                     product.priceAndSeller,
-                                                     product.link,
-                                                     product.shippingPrice);
-
-    // Load the content using jQuery's append.
-    $productContainer.append(productElementHTML);
-
-    // Add the container to the results page, into the corresponding product wrapper.
-    $('#shopping-results-wrapper').append($productContainer);
-  });
+async function appendShoppingResults(shoppingResults) {
+  shoppingResults.forEach((result) => {
+          // Show the user the shopping query built to search on Google Shopping.
```

Comment:
Nit: let's use 4 spaces for this forEach block (here I think are 8)
```suggestion
  shoppingResults.forEach((result) => {
      // Show the user the shopping query built to search on Google Shopping.
```
and so for the next lines

Reply:
I'll run it through the linter once again